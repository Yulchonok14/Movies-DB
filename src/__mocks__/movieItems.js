export const movieItems = [
    {
        id: 111,
        title: 'Bloodshot',
        release_date: '2020-05-01',
        vote_average: 3.6,
        genres: ['Action', 'Drama', 'Sci-Fi'],
        poster_path: 'https://upload.wikimedia.org/wikipedia/ru/9/9d/Bloodshot_poster.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 110
    },
    {
        id: 222,
        title: 'The invisible man',
        release_date: '2019-05-01',
        vote_average: 4.6,
        genres: ['Horror', 'Sci-Fi', 'Thriller'],
        poster_path: 'https://avatars.mds.yandex.net/get-kinopoisk-image/1900788/a90dd0dd-a8ae-4112-9865-23eb8f5879a8/360',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 185
    },
    {
        id: 333,
        title: 'Monster trucks',
        release_date: '2017-06-01',
        vote_average: 3.9,
        genres: ['Adventure', 'Action', 'Comedy'],
        poster_path: 'https://upload.wikimedia.org/wikipedia/ru/5/5e/Monster_Trucks_%28film%29.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 154
    },
    {
        id: 444,
        title: 'Cold comes the night',
        release_date: '2018-05-01',
        vote_average: 4.4,
        genres: ['Drama', 'Thriller', 'Crime'],
        poster_path: 'https://laughingsquid.com/wp-content/uploads/2014/01/CCTN_poster_Billy_HD-for-web.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 129
    },
    {
        id: 555,
        title: 'Snowden',
        release_date: '2020-05-21',
        vote_average: 4.8,
        genres: ['Drama', 'Thriller', 'Biography'],
        poster_path: 'https://upload.wikimedia.org/wikipedia/ru/1/1a/Snowden_%28film%29.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 113
    },
    {
        id: 666,
        title: 'Knock knock',
        release_date: '2011-05-01',
        vote_average: 2.6,
        genres: ['Horror', 'Thriller', 'Adventure'],
        poster_path: 'https://i1.wp.com/morbidlybeautiful.com/wp-content/uploads/2015/10/kk-2.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 187
    }
];