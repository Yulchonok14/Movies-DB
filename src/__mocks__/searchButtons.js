export const searchByButtonsArr = [
    {id: 111, text: 'title', label: 'title', isActive: true},
    {id: 222, text: 'genres', label: 'genre', isActive: false}
];