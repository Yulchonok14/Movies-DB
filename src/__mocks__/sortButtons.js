export const sortByButtonsArr = [
    {id: 111, text: 'release_date', label: 'release date', isActive: true},
    {id: 222, text: 'vote_average', label: 'rating', isActive: false}
];