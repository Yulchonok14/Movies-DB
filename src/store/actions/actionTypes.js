export const GET_MOVIES_HUNTER = 'GET_MOVIES_HUNTER';
export const SEARCH_FILTER = 'SEARCH_FILTER';
export const SORT_FILTER = 'SORT_FILTER';
export const SORT_MOVIES = 'SORT_MOVIES';

export const GET_MOVIES_DETAILS = 'GET_MOVIES_DETAILS';
export const CHOOSE_MOVIE_DETAILS = 'CHOOSE_MOVIE_DETAILS';
export const PASS_CHOSEN_MOVIE = 'PASS_CHOSEN_MOVIE';
