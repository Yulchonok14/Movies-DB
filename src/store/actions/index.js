export {
    searchMovieDetails,
    chooseMovieDetails,
    passChosenMovie,
    getDetailsMovies
} from './movieDetailsActions';
export {
    searchMovieHunter,
    searchFilter,
    sortFilter,
    sortMovies,
    getHunterMovies
} from './movieHunterActions'