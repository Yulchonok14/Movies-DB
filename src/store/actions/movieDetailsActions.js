import * as actionTypes from './actionTypes';
import { getQueryString } from '../utils/storeUtils';

export const searchMovieDetails = (searchText) => {
    let url = 'https://reactjs-cdp.herokuapp.com/movies';
    const params = {
        search: searchText,
        searchBy: 'genres'
    };
    url += '?' + getQueryString(params);
    return dispatch => {
        return fetch(url)
            .then(response => response.json())
            .then(movieItems => {
                dispatch(getDetailsMovies(movieItems.data))
            } )
    }
};

export const getDetailsMovies = (movieItems) => {
    return {
        type: actionTypes.GET_MOVIES_DETAILS,
        payload: {
            movieItems: movieItems
        }
    }
};

export const chooseMovieDetails = (id) => {
    return {
        type: actionTypes.CHOOSE_MOVIE_DETAILS,
        payload: {id: id}
    }
};

export const passChosenMovie = (movie) => {
    return {
        type: actionTypes.PASS_CHOSEN_MOVIE,
        payload: {chosenMovie: movie}
    }
};