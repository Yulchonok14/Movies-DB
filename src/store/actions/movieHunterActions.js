import * as actionTypes from './actionTypes';
import { getQueryString } from '../utils/storeUtils';

export const searchMovieHunter = (searchText, searchBy) => {
    if(searchText) {
        let url = 'https://reactjs-cdp.herokuapp.com/movies';
        const params = {
            search: searchText,
            searchBy: searchBy
        };
        url += '?' + getQueryString(params);
        return dispatch => {
            return fetch(url)
                .then(response => response.json())
                .then(movieItems => {
                    dispatch(getHunterMovies(movieItems.data, searchText))
                })
        }
    } else {
        return dispatch => dispatch(getHunterMovies([], null));
    }
};

export const getHunterMovies = (movieItems, searchText) => {
    return {
        type: actionTypes.GET_MOVIES_HUNTER,
        payload: {
            movieItems: movieItems,
            searchText: searchText
        }
    }
};

export const searchFilter = (searchFilterText) => {
    return {
        type: actionTypes.SEARCH_FILTER,
        payload: {searchFilterText: searchFilterText}
    }
};

export const sortFilter = (sortFilterText) => {
    return {
        type: actionTypes.SORT_FILTER,
        payload: {sortFilterText: sortFilterText}
    }
};

export const sortMovies = () => {
    return {
        type: actionTypes.SORT_MOVIES
    }
};