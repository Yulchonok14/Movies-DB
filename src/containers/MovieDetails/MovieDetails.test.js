import React from 'react';

import {mount, configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk';

import ConnectedMovieDetails, { MovieDetails, mapDispatchToProps } from './MovieDetails';
import * as actionCreators from '../../store/actions/index';
import * as actionTypes from '../../store/actions/actionTypes';
import { movieItems } from '../../__mocks__/movieItems';


configure({
    adapter: new Adapter()
});

describe('ConnectedMovieDetails', () => {
    const initialState = {
        details: {
            movieDetailsItems: [],
            chosenMovieItem: movieItems[0]
        }
    };
    const mockStore = configureStore([thunk]);
    jest.spyOn(ConnectedMovieDetails.WrappedComponent.prototype, 'componentDidMount').mockImplementation();

    let store, wrapper;

    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<ConnectedMovieDetails store={store}/>);
    });

    it('check Prop matches with initialState', () => {
        expect(wrapper.find(MovieDetails).prop('movieItems')).toEqual(initialState.details.movieDetailsItems);
        expect(wrapper.find(MovieDetails).prop('chosenMovieItem')).toEqual(initialState.details.chosenMovieItem);
    });
});

describe('<MovieDetails/>', () => {
    const initialState = {
        details: {
            movieDetailsItems: [],
            chosenMovieItem: movieItems[0]
        }
    };
    const mockStore = configureStore([thunk]);
    const testFunction = () => true;
    const onMovieSearchedSpy = jest.fn();
    const onMovieSearchedByIdSpy = jest.fn();
    let store, wrapper;

    beforeEach(() => {
        store = mockStore(initialState);

        wrapper = shallow(<MovieDetails
            movieItems={[]}
            onMovieSearched={onMovieSearchedSpy}
            onMovieSearchedById={onMovieSearchedByIdSpy}
            chosenMovieItem={movieItems[0]}
            onMovieChosen={testFunction}/>, {disableLifecycleMethods:true});
    });

    it('render the connected component', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('check CHOOSE_MOVIE_DETAILS on dispatching', () => {
        let action;
        store.dispatch(actionCreators.chooseMovieDetails(111));
        action = store.getActions();
        expect(action[0].type).toBe(actionTypes.CHOOSE_MOVIE_DETAILS);
    });

    it('capturing Snapshot of MovieDetails', () => {
        const testFunction = () => true;
        const renderedValue =  renderer.create(<MovieDetails
            movieItems={initialState.details.movieDetailsItems}
            chosenMovieItem={initialState.details.chosenMovieItem}
            onMovieChosen={testFunction}
            onMovieSearchedById={testFunction}
            match={{params: {}}}
            history={{listen: testFunction}}/>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });

    it('check GET_MOVIES_DETAILS on dispatching', async () => {
        window.fetch = jest.fn().mockImplementation(() =>
            Promise.resolve({
                json: () => Promise.resolve({ data: [] })}));
        await store.dispatch(actionCreators.searchMovieDetails('test'));
        const action = store.getActions();
        expect(action[0].type).toBe(actionTypes.GET_MOVIES_DETAILS);
    });

    it('test mapDispatchToProps', () => {
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).onMovieChosen(111);
        expect(dispatch.mock.calls[0][0]).toEqual({type: actionTypes.CHOOSE_MOVIE_DETAILS, payload:{'movieItem': 111}});

        mapDispatchToProps(dispatch).onMovieSearched('test');
        expect(dispatch.mock.calls[1][0]).toBeDefined();
    });

    it('componentDidUpdate dispatch GET_MOVIES_DETAILS action', () => {
        let prevProp = {chosenMovieItem: movieItems[0]};
        wrapper.instance().componentDidUpdate(prevProp);
        expect(onMovieSearchedSpy).not.toHaveBeenCalled();

        prevProp = {chosenMovieItem: movieItems[1]};
        wrapper.instance().componentDidUpdate(prevProp);
        expect(onMovieSearchedSpy).toHaveBeenCalled();
    });
});