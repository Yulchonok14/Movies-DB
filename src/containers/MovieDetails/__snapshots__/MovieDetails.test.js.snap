// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetails/> capturing Snapshot of MovieDetails 1`] = `
Array [
  <div>
    <img
      alt="Selected movie poster."
      onError={[Function]}
      src="https://upload.wikimedia.org/wikipedia/ru/9/9d/Bloodshot_poster.jpg"
    />
    <div>
      <div>
        <span>
          Bloodshot
        </span>
        <span>
          3.6
        </span>
      </div>
      <div>
        <span>
          2020
        </span>
        <label>
          year
        </label>
        <span>
          110
        </span>
        <label>
          min
        </label>
      </div>
      <p>
        Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”
      </p>
    </div>
  </div>,
  <div>
    <span>
      Films by Action genre
    </span>
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <label />
      <div />
    </div>
  </div>,
  <div>
    <div>
      No films found
    </div>
  </div>,
]
`;
